src/libraries/Microsoft.Extensions.Primitives/src/StringSegmentComparer.cs

```diff
@@ -6,11 +6,20 @@
 
 namespace Microsoft.Extensions.Primitives
 {
+    /// <summary>
+    /// Compares two <see cref="StringSegment"/> instances.
```

Comment:
nit: use `objects` instead of `instances`. just to be consistent with the rest of the writing.